src/security-rules/security-rules.ts

```diff
@@ -188,6 +200,21 @@ export class SecurityRules implements FirebaseServiceInterface {
         return this.getRuleset(stripProjectIdPrefix(rulesetName));
       });
   }
+
+  private releaseRuleset(ruleset: string | RulesetMetadata, releaseName: string): Promise<void> {
+    if (!validator.isNonEmptyString(ruleset) &&
+      (!validator.isNonNullObject(ruleset) || !validator.isNonEmptyString(ruleset.name))) {
```

Comment:
It's interesting that the name could come in either as a String or as part of the RulesetMetadata Object. Why did you opt for this and not the simpler signature of only passing in the name as a String, making it the caller's job to pluck out the name if it had an Object?

Reply:
We can use the same private method for releasing storage rulesets. By doing it here we save a few lines from being duplicated.